### DISCUSSION TITLE
关于MSYS2和Mingw的编译问题

### DISCUSSION THREAD

**Discussion by alphaDeng** [sogou/workflow#904] [Q&A] [answered]
您好，请问workflow中有没有在MSYS2和Mingw下（Windows平台下）进行编译的方法？
目前我尝试从Cmake（选择MIngw）和MSYS2中的Mingw环境（直接使用make来进行操作）入手，但好像都不成功，出现的问题有：
1. Cmake中直接提示不支持mingw
2. MSYS2中则出现以下的提示：
which: no cmake3 in (/mingw64/bin:/usr/local/bin:/usr/bin:/usr/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl)
mkdir -p build.cmake
cd build.cmake && cmake -D CONSUL= -D KAFKA= -D MYSQL= -D REDIS= -D UPSTREAM= D:/software/programming/msys2/home/compile/workflow-master
CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "/home/compile/workflow-master/build.cmake/CMakeFiles/CMakeOutput.log".
mingw32-make: *** [GNUmakefile:16: base] Error 1

**Comment by Barenboim** [comment, marked as answer]
是的，我们之前放弃mingw和cygwin的编译了。没有人可以维护。
这两个用的是windows API吧？你只能切换到windows分支试试。
